src/render/metal/SDL_render_metal.m

```diff
@@ -1095,20 +1101,24 @@ - (void)dealloc
     for (int i = 0; i < count; i++, points++) {
         *(verts++) = points->x;
         *(verts++) = points->y;
-        *((int *)verts++) = color;
+        *((SDL_Color *)verts++) = color;
```

Comment:
Are we guaranteed that this cast is safe and correct?

Reply:
I'm not sure - at the very least I don't think this will cause any regressions in terms of alignment issues compared to before the PR (however I don't consider myself an expert on C structure alignment across compilers).

I think changing the backend code to use a fuller structure like `SDL_Vertex` (and an equivalent without UVs when there's no texture) is probably a good change for a future PR. But... the external `SDL_RenderGeometryRaw` API is probably a little questionable in general because it can accept arbitrary unaligned strides. Maybe that's a problem for a separate issue report.

Reply:
IIRC alignment requirements of structs are the size of the biggest (primitive type) member.
So I *think* that `SDL_Color` has no alignment requirements because it only contains (single) bytes - while the `int` needed to be aligned to 4 bytes (though x86 doesn't mind unaligned reads, other platforms, like SPARC IIRC, crash when that happens)